Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -1238,11 +1238,15 @@ internal void ServerUpdate()
         {
             if (observers != null && observers.Count > 0)
             {
+                ulong dirtyComponentsMask = GetDirtyMask(false);
+                if (dirtyComponentsMask == 0L)
+                    return;
+
                 // one writer for owner, one for observers
                 using (PooledNetworkWriter ownerWriter = NetworkWriterPool.GetWriter(), observersWriter = NetworkWriterPool.GetWriter())
                 {
                     // serialize all the dirty components and send (if any were dirty)
-                    OnSerializeAllSafely(false, ownerWriter, out int ownerWritten, observersWriter, out int observersWritten);
+                    OnSerializeAllSafely(false, dirtyComponentsMask, ownerWriter, out int ownerWritten, observersWriter, out int observersWritten);
```

Comment:
What am I missing here?   OnSerializeAllSafely was only being called once per NetworkIdentity, so either way we only called dirtyComponentsMask once per NI.

So how does this improve performance?

Reply:
It does the dirty check before getting 2 writers out of the pool

` using (PooledNetworkWriter ownerWriter = NetworkWriterPool.GetWriter(), observersWriter = NetworkWriterPool.GetWriter())`

Just moving the check to before the using increases performance of NetworkManager.LateUpdate from 5.69ms to 4.9ms when nothing is dirty.

The code here atm is just something i through together quickly to test the performance, I'm looking into other ways to check if network identity is dirty so we can benchmark those too